src/CommunityToolkit.Aspire.Hosting.Rust/RustAppHostingExtension.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Aspire.Hosting.ApplicationModel;
+using CommunityToolkit.Aspire.Hosting.Rust.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Rust applications to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class RustAppHostingExtension
+{
+    /// <summary>
+    /// Adds a Rust application to the application model. Executes the executable Rust app.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="workingDirectory">The working directory to use for the command.</param>
+    /// <param name="args">The optinal arguments to be passed to the executable when it is started.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<RustAppExecutableResource> AddRustApp(this IDistributedApplicationBuilder builder, [ResourceName] string name, string workingDirectory, string[]? args = null)
```

Comment:
Let's be similar to go and the node extensions and have working directory as optional, defaulting to a sibling named the same as the resource.

Reply:
Working directory is not optional in go extensions.

https://github.com/CommunityToolkit/Aspire/blob/493c150b4a04ce0cb35d761d34236b9a79b259b4/src/CommunityToolkit.Aspire.Hosting.Golang/GolangAppHostingExtension.cs#L11-L37

Reply:
Oh, never mind then